controller/ca/controller.go

```diff
@@ -172,3 +183,15 @@ func (c *CertificateController) handlePodAdd(obj interface{}) {
 func (c *CertificateController) handlePodUpdate(oldObj, newObj interface{}) {
 	c.handlePodAdd(newObj)
 }
+
+func (c *CertificateController) handleMWCAdd(obj interface{}) {
+	mwc := obj.(*v1beta1.MutatingWebhookConfiguration)
+	log.Debugf("enqueuing secret write for mutating webhook configuration %q", mwc.ObjectMeta.Name)
+	for _, webhook := range mwc.Webhooks {
+		c.queue.Add(fmt.Sprintf("%s.%s.%s", webhook.ClientConfig.Service.Name, pkgK8s.Service, webhook.ClientConfig.Service.Namespace))
```

Comment:
Related to my previous comment, I think we still need to check the name of the MWC here to make sure that the update pertains to the MWC that is being created by the proxy-injector process:

```diff
diff --git a/controller/ca/controller.go b/controller/ca/controller.go
index 34514f95..223c5ff9 100644
--- a/controller/ca/controller.go
+++ b/controller/ca/controller.go
@@ -186,9 +186,11 @@ func (c *CertificateController) handlePodUpdate(oldObj, newObj interface{}) {
 
 func (c *CertificateController) handleMWCAdd(obj interface{}) {
 	mwc := obj.(*v1beta1.MutatingWebhookConfiguration)
-	log.Debugf("enqueuing secret write for mutating webhook configuration %q", mwc.ObjectMeta.Name)
-	for _, webhook := range mwc.Webhooks {
-		c.queue.Add(fmt.Sprintf("%s.%s.%s", webhook.ClientConfig.Service.Name, pkgK8s.Service, webhook.ClientConfig.Service.Namespace))
+	if mwc.Name == pkgK8s.ProxyInjectorWebhookConfig {
+		log.Debugf("enqueuing secret write for mutating webhook configuration %q", mwc.ObjectMeta.Name)
+		for _, webhook := range mwc.Webhooks {
+			c.queue.Add(fmt.Sprintf("%s.%s.%s", webhook.ClientConfig.Service.Name, pkgK8s.Service, webhook.ClientConfig.Service.Namespace))
+		}
 	}
 }
 ```